zaza/openstack/charm_tests/hacluster/tests.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+# Copyright 2018 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""HACluster testing."""
+
+import logging
+import os
+
+import zaza.openstack.charm_tests.test_utils as test_utils
+import zaza.openstack.utilities.juju as juju_utils
+import zaza.openstack.configure.hacluster
+
+from . import utils as hacluster_utils
+
+
+class HaclusterTest(test_utils.OpenStackBaseTest):
+    """hacluster tests."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for running hacluster tests."""
+        super(HaclusterTest, cls).setUpClass()
+        cls.vip = os.environ.get("TEST_VIP00")
+
+    def test_900_action_cleanup(self):
+        """The services can be cleaned up. """
+        status = zaza.model.get_status().applications[self.application_name]
+
+        # libjuju juju status no longer has units for subordinate charms
+        # Use the application it is subordinate-to to check workload status
+        if status.get("units") is None and status.get("subordinate-to"):
+            primary_status = juju_utils.get_application_status( \
+                    status.get("subordinate-to")[0])
+            leader = None
+            for unit in primary_status["units"]:
+                if primary_status["units"][unit].get('leader'):
+                    leader = unit
+        leader_unit = zaza.model.get_unit_from_name(leader)
+        assert leader_unit.workload_status == "active"
```

Comment:
We don't need this check. And in fact, you can remove the leader_unit as we never use it for anything else.